pkg/apis/crds/karpenter.k8s.aws_ec2nodeclasses.yaml

```diff
@@ -271,15 +338,39 @@ spec:
                       description: Tags is a map of key/value tags used to select
                         subnets Specifying '*' for a value selects all values for
                         a given tag key.
+                      maxProperties: 30
                       type: object
+                      x-kubernetes-validations:
+                      - message: tags cannot be empty
+                        rule: self.size() != 0
+                      - message: empty tag keys and values aren't supported
+                        rule: self.all(k, k != '' && self[k] != '')
                   type: object
+                maxItems: 20
                 type: array
+                x-kubernetes-validations:
+                - message: subnetSelectorTerms cannot be empty
+                  rule: self.size() != 0
+                - message: '''id'' is mutually exclusive, cannot be set with a combination
+                    of other fields in'
+                  rule: '!self.all(x, has(x.id) && has(x.tags))'
               tags:
                 additionalProperties:
                   type: string
                 description: Tags to be applied on ec2 resources like instances and
                   launch templates.
                 type: object
+                x-kubernetes-validations:
+                - message: empty tag keys aren't supported
+                  rule: self.all(k, k != '')
+                - message: tag contains in restricted tag matching kubernetes.io/cluster/
+                  rule: self.all(k, !k.startsWith('kubernetes.io/cluster') )
+                - message: tag contains in restricted tag matching karpenter.sh/provisioner-name
+                  rule: self.all(k, !k.startsWith('karpenter.sh/provisioner-name'))
```

Comment:
I think everything here except for the `kubernetes.io/cluster` key are equals checks rather than a `startsWith` check